File: solutions/scala/problem-00019.scala
import java.util.Calendar
import java.util.Calendar._

var cal = Calendar.getInstance
var sundays = 0
(1901 to 2000).foreach(year => {
  (0 to 11).foreach(month => {
    cal.set(year, month, 1) 
    if (cal.get(DAY_OF_WEEK) == SUNDAY)
      sundays += 1
  })
})
println(sundays)

File: solutions/ruby/problem-00019.rb
#!/usr/bin/env ruby
# [minhuang@mocha:ruby]$ time ruby problem-00019.rb 
# 171
#
# real  0m0.174s
# user  0m0.160s
# sys 0m0.000s
require 'date'

sundays = 0
(1901..2000).each do |year|
  (1..12).each do |month|
    d = Date.new(year, month, 1)
    sundays += 1 if d.cwday == 7 # Sunday
  end
end

puts sundays
